lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -13,21 +14,24 @@
 from .cmake import CMakeBuilder, CMakePackage
 
 
-def cmake_cache_path(name, value, comment=""):
+def cmake_cache_path(name, value, comment="", force=True):
     """Generate a string for a cmake cache variable"""
-    return 'set({0} "{1}" CACHE PATH "{2}")\n'.format(name, value, comment)
+    force_str = "FORCE" if force else ""
+    return 'set({0} "{1}" CACHE PATH "{2}" {3})\n'.format(name, value, comment, force_str)
 
 
-def cmake_cache_string(name, value, comment=""):
+def cmake_cache_string(name, value, comment="", force=True):
     """Generate a string for a cmake cache variable"""
-    return 'set({0} "{1}" CACHE STRING "{2}")\n'.format(name, value, comment)
+    force_str = "FORCE" if force else ""
+    return 'set({0} "{1}" CACHE STRING "{2}" {3})\n'.format(name, value, comment, force_str)
 
 
-def cmake_cache_option(name, boolean_value, comment=""):
+def cmake_cache_option(name, boolean_value, comment="", force=True):
     """Generate a string for a cmake configuration option"""
 
     value = "ON" if boolean_value else "OFF"
-    return 'set({0} {1} CACHE BOOL "{2}")\n'.format(name, value, comment)
+    force_str = "FORCE" if force else ""
```

Comment:
@scheibelp So the behavior that I experienced is that if there is an existing `CMakeCache.txt` file then without the force, the config file's behavior is ignored.  I will check to see how command line arguments will interact.

Reply:
AFAIK `CMakeCache.txt` is typically not distributed with the source, I wanted to check if that's also true for `lbann` (or any other packages you are building).

(this could still be an issue in Spack if you restart after a failed install and changing the package.py to generate a different `.cmake` file, but I assume that would be rare)

Reply:
@scheibelp Here is a response from Tom. 
```
If you do
cmake -DSOME_VAR=ON_or_OFF -C my_cache.cmake ...
then the FORCE will cause SOME_VAR to have the value in my_cache.cmake. However,
cmake -C my_cache.cmake -DSOME_VAR=ON_or_OFF ...
will cause SOME_VAR to have the value specified on the command line.
Do note that this was experimentally determined using CMake 3.22.1 with all CMake policies set to match that version. 
```

Reply:
I will change the default so that it does not set the `FORCE` flag.